invision-app/app/components/movieList/movieList.js

```diff
@@ -1,19 +1,56 @@
-import React from 'react';
+import React,{useEffect} from 'react';
 import './movieList.scss'
 import { MovieCard } from '../movieCard';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { getMovies } from '../../store/actions';
+import { MoivesFound } from '../moviesFound/moviesFound';
 
-export function MovieList(props) {
-    return (
-        <div className="movies-list">
-            {props.movies.map(mov => <div className="movies-list__item" key={mov.id}>
-                <MovieCard movie={mov} key={mov.id} updateMovie={props.updateMovie} deleteMovie={props.deleteMovie} showMovieDetails={props.showMovieDetails}/></div>)}
-        </div>
-    );
-}   
-MovieList.propTypes = {   
-    deleteMovie: PropTypes.func.isRequired,
-    updateMovie: PropTypes.func.isRequired,
+const MovieList=({ getMovies, customMovies, sortedMovies, showMovieDetails })=> {
+
+  useEffect(() => {
+    getMovies();
+  }, []);
+
+  return (   
+   <div>
+     <MoivesFound count={sortedMovies.length === 0?customMovies.length:sortedMovies.length}/>    
   
+    <div className="movies-list">
+         
+      {
+        sortedMovies.length === 0 ?
```

Comment:
it is too complicated solution. Your sort functions, such as SORT_BY_RELEASE_DATE, update **movies** list and it works correctly until filter the movies. After that sorting does not work because **sortedMovies** is not empty any more. I'd reccomend to have only one list of movies to display. Based on sort and filter values you can change this list on the redux store.

Reply:
Another example: 
1. Filter movies list
2. Edit on of the movies - nothing changes on the oage because only **movies** list was changed, but ** sortedMovies** is displayed